src/java.base/share/native/launcher/main.c

```diff
@@ -62,6 +62,11 @@ main(int argc, char **argv)
     char** jargv;
     const jboolean const_javaw = JNI_FALSE;
 #endif /* JAVAW */
+
+#if !defined(WINDOWS) && !defined(MACOSX)
+   JLI_ReExecLauncher(argc, argv);
+#endif
```

Comment:
The naming is problematic here as it looks like it always re-execs on Unix/Linux systems. I think what you are looking for is something like ReExecLauncherIfNeeded or something that make it clear that it doesn't re-exec always.